extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/DataSourceDecorator.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.flyway.runtime;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import javax.sql.DataSource;
+
+public class DataSourceDecorator implements DataSource {
```

Comment:
The name doesn't really explain what this class does, but I am not very good at naming, so I'll let others decide about it